### PR TITLE
Add content on workflow approval

### PR DESCRIPTION
**Pull request by Nashaath**
## Purpose
Add content on workflow approval.

Screenshot 2024-10-31 at 08 07 53

Screenshot 2024-10-31 at 07 42 04

Screenshot 2024-10-31 at 08 49 27

### CODE REVIEW

**Review by LakshanSS — COMMENTED** [review]
We have decided to provide this feature only for paid users.
Should we say anything related to that in the documentation?

File: en/docs/administer/configure-approvals-for-choreo-workflows.md
Comment (abeykoon) [inline en/docs/administer/configure-approvals-for-choreo-workflows.md:29]: submit a request to use the workflow -> submit a request for approval
for a developer to use the workflow -> for a developer to proceed with the task related to the workflow.

File: en/docs/administer/review-workflow-approval-requests.md
Comment (abeykoon) [inline en/docs/administer/review-workflow-approval-requests.md:8]: `Other members in the organization`. In fact not any member.  It should be one configured to approve workflows

**Comment by abeykoon** [comment]
There are a few changes in the permission model related to this feature. Let me brief them below. 

Notifications

1. Email will go to the mentioned direct assignees 
2. As assignees in workflow config one or more roles can be selected. People who are in user groups bound to those roles at the project level of the project in which workflow request is being raised will be notified. 

What this means is, if you are project manager (role) of project A (bound project), you will get notified for workflow requests made within project A, nothing else. 

If user has org level permissions, he/she is notified for all the requests in any project. 

Permissions to Approve

1. User need to have relevant permission under `WORKFLOW-MANAGEMENT` . User will be able to view and approve workflows with permissions only. 
2. To be able to approve user MUST have `PROJECT-MANAMEMENT` --> `project management`

File: en/docs/administer/review-workflow-approval-requests.md
Comment (Nashaath) [reply en/docs/administer/review-workflow-approval-requests.md]: Done

**Comment by Nashaath** [comment]
> We have decided to provide this feature only for paid users. Should we say anything related to that in the documentation?

I have not added a note to indicate this because this decision is not yet finalized.

File: en/docs/administer/review-workflow-approval-requests.md
Comment (Nashaath) [inline en/docs/administer/review-workflow-approval-requests.md:35]: @abeykoon this limitation is not there now right?

File: en/docs/administer/review-workflow-approval-requests.md
Comment (Nashaath) [inline en/docs/administer/review-workflow-approval-requests.md:5]: @abeykoon is this technically correct?

File: en/docs/administer/review-workflow-approval-requests.md
Comment (Nashaath) [inline en/docs/administer/review-workflow-approval-requests.md:12]: @abeykoon Please have a look at the note content and let me know in case anything needs to change.

File: en/docs/administer/configure-approvals-for-choreo-workflows.md
Comment (abeykoon) [inline en/docs/administer/configure-approvals-for-choreo-workflows.md:22]: We need to tweak this a bit. Even any `role` and `assignee` can be selected, they can view and approve the requests only if they have `required permissions`.

Required permissions are
1. `Project Manage`
2. `Relevant WorkflowPermission`

I think we are describing the permissions required somewhere. We can point to that header from here.

File: en/docs/administer/configure-approvals-for-choreo-workflows.md
Comment (abeykoon) [inline en/docs/administer/configure-approvals-for-choreo-workflows.md:25]: This is not always true. For some workflows (like for subscription apporvals coming soon), once manager approves, the requested change will get applied in Choreo. But for `env promotions`, what is mentioned above is correct.

File: en/docs/administer/review-workflow-approval-requests.md
Comment (abeykoon) [inline en/docs/administer/review-workflow-approval-requests.md:13]: project manager - > Project Admin. 
That's the new default role we are gonna provision soon.

Comment (abeykoon) [reply en/docs/administer/review-workflow-approval-requests.md:13]: Is there a doc link for project level group-role assignments? If so, we can link here.

File: en/docs/administer/review-workflow-approval-requests.md
Comment (abeykoon) [inline en/docs/administer/review-workflow-approval-requests.md:15]: You will see them in `Past Approvals` tab.

File: en/docs/administer/review-workflow-approval-requests.md
Comment (abeykoon) [inline en/docs/administer/review-workflow-approval-requests.md:16]: use the workflow means, he can continue to perform the task to which the approval is granted.

File: en/docs/administer/review-workflow-approval-requests.md
Comment (abeykoon) [inline en/docs/administer/review-workflow-approval-requests.md:24]: Now there are two tabs. `Pending` will show pending approvals to review. `Past` will show already reviewed approval requests by you or someone else who is eligible to review the request. Also the requests cancelled by the requestors are shown in the same `Past` approvals list.

**Comment by Nashaath** [comment]
@abeykoon we have listed the predefined roles and groups in Choreo here [1]. Might have to update accordingly to include the new roles introduced via this feature right? Please let me know the exact role/group names to include.

[1] https://wso2.com/choreo/docs/choreo-concepts/organization/#manage-user-permission

File: en/docs/administer/review-workflow-approval-requests.md
Comment (abeykoon) [reply en/docs/administer/review-workflow-approval-requests.md:35]: Yes, it is no longer there. As described above `Past` tab will contain all already reviewed and requestor cancelled workflow requests.

File: en/docs/develop-components/request-for-workflow-approval.md
Comment (abeykoon) [inline en/docs/develop-components/request-for-workflow-approval.md:14]: `environment promotion` is enabled only when you are deploying to a `critical` environment.

File: en/docs/administer/review-workflow-approval-requests.md
Comment (abeykoon) [inline en/docs/administer/review-workflow-approval-requests.md:33]: You also can add a note for the requestor.

File: en/docs/develop-components/request-for-workflow-approval.md
Comment (abeykoon) [inline en/docs/develop-components/request-for-workflow-approval.md:7]: We also need to include cancel capability.

**Review by abeykoon — APPROVED** [review]

**Review by uvindra — APPROVED** [review]

**Pull request merged**
